### ISSUE TITLE
[FR] Provide current benchmark information

### ISSUE THREAD

**Issue by SunBlack** [google/benchmark#1677]
**Is your feature request related to a problem? Please describe.**
Currently at GoogleTest we use the name of the test to create a unique folder per test where the tests should write (so that the folder names are unique, but also easy to find when debugging for example). So we are using the data provided by `testing::UnitTest::GetInstance()->current_test_info()` in a helper class. Sadly, there is nothing similar in the Benchmark library.

**Describe the solution you'd like**
Provide a way to retrieve information about the current running benchmark - similar to [`TestParamInfo` of GoogleTest](http://google.github.io/googletest/reference/testing.html#TestParamInfo).

**Comment by EfesX** [comment]
Am I wrong, or does `benchmark::State` already provide all the required information about the benchmark?

**Comment by dmah42** [comment]
yes i'm surprised that `State` doesn't hold what you'd need as it's used to output this information in the reports.

**Comment by LebedevRI** [comment]
I suppose we could easily add `family_index`/`per_family_instance_index`/`repetition_index` to `State` (currently they are only in `Run`).
